### PR TITLE
Oss components : upgrade column selection panel

### PR DESCRIPTION
**Pull request by JulienVannier66**
Related to: https://github.com/upfluence/backlog/issues/312
### What does this PR do?

When the user opens the workflow column selector, it does not close if the user clicks outside this section. It is necessary to close the column selector if the user clicks outside this section (focus out).

### What are the observable changes?
Peek 2021-02-16 16-06

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [ ] ~Added/updated tests~
- [ ] ~Added/updated documentation~
- [ ] ~Migrated touched components to Glimmer Components~
- [x] Properly labeled

### CODE REVIEW

File: app/templates/components/column-visibility-panel.hbs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      {{#unless column.unhideable}}
+        <div class="margin-bottom-sm">
+          {{#upf-checkbox value=column.visible hasLabel=true}}
+            <span class="text-color-contrast">{{column.title}}</span>
+          {{/upf-checkbox}}
+        </div>
+      {{/unless}}
+    {{/each}}
+  </div>
+{{/if}}
```
Comment (tgallice) [inline app/templates/components/column-visibility-panel.hbs:23]: missing endline

File: app/templates/components/column-visibility-panel.hbs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<button class="upf-btn upf-btn--default upf-btn--small upf-link--reset button-column-visibility-panel">
+  <i class="fa fa-columns"></i> &nbsp;
+  {{#if displayedColumnsPanel}}
+    <i class="fa fa-caret-up"></i>
+  {{else}}
+    <i class="fa fa-caret-down"></i>
+  {{/if}}
```
Comment (tgallice) [inline app/templates/components/column-visibility-panel.hbs:7]:
```suggestion
<i class="fa {{if displayedColumnsPanel 'fa-caret-up' 'fa-caret-down'}}"></i>
```

File: addon/components/column-visibility-panel.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+export default Component.extend({
+  displayedColumnsPanel: false,
+  hasToggleableColumns: false,
+
+  handleClick(event) {
+    if (event.target.closest('.button-column-visibility-panel') != null) {
+      this.toggleProperty('displayedColumnsPanel');
+    }
+    else if (event.target.closest('.side-panel--appearance') == null) {
```
Comment (tgallice) [inline addon/components/column-visibility-panel.js:11]:
```suggestion
    } else if (!event.target.closest('.side-panel--appearance')) {
```

File: addon/components/column-visibility-panel.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import Component from '@ember/component';
+
+export default Component.extend({
+  displayedColumnsPanel: false,
+  hasToggleableColumns: false,
+
+  handleClick(event) {
+    if (event.target.closest('.button-column-visibility-panel') != null) {
```
Comment (tgallice) [inline addon/components/column-visibility-panel.js:8]:
```suggestion
    if (event.target.closest('.button-column-visibility-panel')) {
```

File: addon/components/column-visibility-panel.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    }
+    else if (!event.target.closest('.side-panel--appearance')) {
+      this.set('displayedColumnsPanel', false);
+    }
+  },
+
+  init(){
+    this._super();
+
+    this.set('bindClickHandler', this.clickHandler.bind(this));
```
Comment (phndiaye) [inline addon/components/column-visibility-panel.js:19]: my bad i meant `this.set('clickHandler', this.handleClick.bind(this))`

Comment (tgallice) [reply addon/components/column-visibility-panel.js]: emberjs component have event based method maybe we can simply use it ?
```js
click(event) {
    if (event.target.closest('.button-column-visibility-panel')) {
      this.toggleProperty('displayedColumnsPanel');
    } else if (!event.target.closest('.side-panel--appearance')) {
      this.set('displayedColumnsPanel', false);
    }
}
```

Comment (tgallice) [reply addon/components/column-visibility-panel.js]: My bad, the click is on the document :) forget my previous message

**Review by phndiaye — APPROVED** [review]

**Review by tgallice — APPROVED** [review]

**Pull request merged**